repository/client.repo.go

```diff
@@ -55,3 +55,18 @@ func (r *ClientRepository) Find(client *models.Client) error {
 
 	return dbTables.Where(&client).First(&client).Error
 }
+
+func (r *ClientRepository) Create(client *models.Client) error {
+	return r.db.Model(&client).Create(&client).First(&client).Error
+}
+
+func (r *ClientRepository) Update(client *models.Client) error {
+	originalClient := models.Client{Id: client.Id}
+	r.db.Model(&originalClient).First(&originalClient)
+
+	return r.db.Model(&originalClient).Save(&client).Error
+}
+
+func (ldr *ClientRepository) Destroy(client *models.Client) error {
+	return ldr.db.Table("clients").Delete(&client).Error
```

Comment:
cái này cần phải cho vào transaction, update toàn bộ `project` sử dụng `client` này cho `clientId` về null.
có thể dùng hook